blogs/2025/11/03/universe-recap.md

```diff
@@ -0,0 +1,118 @@
+
+![Copilot chat in VS Code with Plan agent dropdown highlighted, planning guidance beside dark theme editor, label reads Plan for a focused tone.](planning-agent.png)
+
+When working on medium to large tasks, having a detailed plan beats an open-ended prompt every time. The less info the model has, the more it guesses. The more it guesses, the more mistakes it makes. Prompt Engineering is the fine art of giving the model the answer that you want.
+
+The new Plan agent helps create a detailed plan from lazy prompts like "add drag and drop". That's an actual prompt I sent yesterday. No mention of what to add it to, what page, or whether to use a library. I do this a lot, and I bet I'm not alone.
+
+With the Plan agent, Copilot asks the questions that need answers. It even recommends libraries for drag and drop and gives reasons to pick one over another.
+
+![Plan agent breaking down drag-and-drop into steps recommending React Beautiful DnD and React DnD with comparisons.](plan-in-action.png)
+
+You can answer these with quick replies on separate lines so it knows which answer goes to which question. Here's how I'd answer:
+
+```
+dnd-kit
+yes - what kind of a question is this in 2025
+link creation only
+```
+
+> Pro tip: Change the "workbench.action.chat.submit" keybinding to "Ctrl + Enter" so you stop accidentally sending messages when you just want a new line. Your swear jar will thank you.
+
+When the Plan agent has enough info, it stops asking questions and asks if you're ready to proceed. You can use the new "[Handoff](https://code.visualstudio.com/updates/v1_105#_handoffs)" feature in chat to either proceed or open the full plan in the editor.
+
+![Screenshot showing the Handoff feature in Copilot chat with options to proceed with implementation or open the plan in the editor.](handoffs.png)
+
+Try different models to see which you like best for planning. We've found the [Claude models](https://www.anthropic.com/claude) are great at identifying missing context and edge cases, and asking the right questions.
+
+If you're like me, you'll want to know how the Plan agent works so you can up your prompt engineering game. You can read the Plan prompt by choosing "Configure Agents" from the Command Palette and selecting Plan. It's a great baseline for creating your own [custom agents](https://code.visualstudio.com/updates/v1_105#_support-for-nested-agentsmd-files-experimental). I used it to create one called ["Research"](https://gist.github.com/burkeholland/919d655ae4df5c809b549632c3afb144) that recursively does internet research and writes up its findings.
+
+> Pro tip: You can find hundreds of custom instructions, prompt files and agents over on the [awesome-copilot](https://github.com/github/awesome-copilot) repo. If you haven't checked that out yet, you're missing out. It's a treasure trove of inspiration and ready-made prompts.
```

Comment:
An additional cool thing is that you can use these same agents in the CLI and with Cloud Agent as well when you assign or delegate a task on github.com. Better together